compiler/generator/jax/jax_code_container.cpp

```diff
@@ -0,0 +1,486 @@
+    tab(n+1, *fOut);  
+    *fOut << "return a - y*((a > y/2).astype(jnp.int32))";
+    tab(n, *fOut);  
+
+    // Merge sub containers
+    mergeSubContainers();
+
+    // Functions
+    tab(n, *fOut);
+    gGlobal->gJAXVisitor->Tab(n);
+       
+    tab(n, *fOut);
+    *fOut << "class " << fKlassName << "(nn.Module):";
+    tab(n + 1, *fOut);
+
+    // Fields
+    gGlobal->gJAXVisitor->Tab(n + 1);
+
+    tab(n + 1, *fOut);
+    *fOut << "sample_rate: int";
+
+    tab(n + 1, *fOut);
+    gGlobal->gJAXVisitor->Tab(n);
+
+    tab(n + 1, *fOut);
+    produceInfoFunctions(n + 1, "", "self", false, FunTyped::kDefault, gGlobal->gJAXVisitor);
+    
+    *fOut << "def classInit(self, sample_rate, x, T):";
+    {
+        tab(n + 2, *fOut);
+        *fOut << "state = {}";
+        tab(n + 2, *fOut);
+        tab(n + 2, *fOut);
+        *fOut << "# global declarations:";
+        JAXInitFieldsVisitor initializer(fOut, n + 2);
+        generateDeclarations(&initializer);
+        // Generate global variables initialisation
+        for (const auto& it : fGlobalDeclarationInstructions->fCode) {
+            if (dynamic_cast<DeclareVarInst*>(it)) {
+                it->accept(&initializer);
+            }
+        }
+        tab(n + 2, *fOut);
+        tab(n + 2, *fOut);
+        *fOut << "# init constants";
+        tab(n + 2, *fOut);
+        gGlobal->gJAXVisitor->Tab(n + 2);
+        inlineSubcontainersFunCalls(fInitInstructions)->accept(gGlobal->gJAXVisitor);
+        tab(n + 2, *fOut);
+        *fOut << "# inline subcontainers:";
+        tab(n + 2, *fOut);
+        gGlobal->gJAXVisitor->Tab(n + 2);
+        inlineSubcontainersFunCalls(fStaticInitInstructions)->accept(gGlobal->gJAXVisitor);
+        tab(n + 2, *fOut);
+        *fOut << "# instance clear:";
+        tab(n + 2, *fOut);
+        generateClear(gGlobal->gJAXVisitor);
+        tab(n + 2, *fOut);
+        *fOut << "# build user interface parameters";
+        tab(n + 2, *fOut);
+        *fOut << "state = self.build_interface(state, x, T)";
+        tab(n + 2, *fOut);
+        tab(n + 2, *fOut);
+        *fOut << "# convert numpy array to jax numpy array";
+        tab(n + 2, *fOut);
+        *fOut << "state = jax.tree_map(jnp.array, state)";
+        tab(n + 2, *fOut);
+        tab(n + 2, *fOut);
+        *fOut << "return state";
+        tab(n + 1, *fOut);
+    }
+
+    back(1, *fOut);
+
+    tab(n+1, *fOut);
+    
+    // JSON generation
+    tab(n+1, *fOut);
+    *fOut << "def getJSON(self):";
+    {
+        string json;
+        if (gGlobal->gFloatSize == 1) {
+            json = generateJSON<float>();
+        } else {
+            json = generateJSON<double>();
+        }
+        tab(n + 2, *fOut);
+        *fOut << "json_str = \"\"\"" << flattenJSONforPython(json) << "\"\"\"";
+        tab(n + 2, *fOut);
+        *fOut << "return json.loads(json_str)";
+        tab(n + 1, *fOut);
+    }
+
+    tab(n + 1, *fOut);
+    *fOut << "def load_soundfile(self, filepath):";
```

Comment:
General remark on the necessity to have the backend generate `load_soundfile/add_soundfile/add_nentry/add_button/add_slider`

Reply:
Should be resolved by recent commits.